lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -34,29 +29,45 @@ function isRefreshError(data) {
  *
  * This is used to avoid triggering concurrent refresh requests for the same
  * external API.
- *
- * @type {Map<string, Promise<void>>}
  */
-const activeRefreshCalls = new Map();
+const activeRefreshCalls = new Map<string, Promise<void>>();
+
+export type APICallOptions = {
+  /** Session authorization token. */
+  authToken: string;
+
+  /**
+   * The `/api/...` path of the endpoint to call.
+   *
+   * If the path contains parameters, use {@link urlPath} to generate this.
+   */
+  path: string;
+
+  /** Query parameters. */
+  params?: Record<string, string>;
+
+  /** JSON-serializable body of request. */
+  data?: object;
+
+  /** HTTP method to use. Defaults to POST if `data` is defined or GET otherwise. */
```

Comment:
Any value in typing this as `'POST' | 'GET'`? Or is that unnecessarily restrictive?